crates/editor/src/editor.rs

```diff
@@ -4977,10 +4977,16 @@ impl Editor {
         if !revert_changes.is_empty() {
             self.transact(cx, |editor, cx| {
                 editor.buffer().update(cx, |multi_buffer, cx| {
-                    for (buffer_id, buffer_revert_ranges) in revert_changes {
+                    for (buffer_id, changes) in revert_changes {
                         if let Some(buffer) = multi_buffer.buffer(buffer_id) {
                             buffer.update(cx, |buffer, cx| {
-                                buffer.edit(buffer_revert_ranges, None, cx);
+                                buffer.edit(
+                                    changes.into_iter().map(|(range, text)| {
+                                        (range, text.to_string().map(Arc::<str>::from))
```

Comment:
Buffer API accepts new text in edits as `Arc<str>`, so we now have to convert ropes into this when reverting the text.
Does not seem bad (still better than doing a String::copy on the entire diff base), but let me know if there's a concern.